### PR TITLE
Integrating souls into the underworld

### PR DESCRIPTION
**Pull request by SabrinaTheCookie**
MVP Gameplay Candidate.

- Main Bugs: SpawnManager gets uncontrollable.
- Can ferry souls
- Can return to pickup souls
- Obstacles to be avoided

### CODE REVIEW

File: Assets/Scripts/Boat/BoatCapacity.cs [hunk 1]
```diff
@@ -34,19 +87,24 @@ void Start()
...
     {
-        currentCapacity = currentCapacity - amount;
-        if (currentCapacity == 0)
+        CurrentCapacity = CurrentCapacity - amount;
+        if (CurrentCapacity == 0)
         {
             OnBoatDestroyed?.Invoke();
         }
-        else if (currentCapacity < 0)
+        else if (CurrentCapacity < 0)
```
Comment (TomasMorton) [inline Assets/Scripts/Boat/BoatCapacity.cs:105]: I'm a bit confused by this logic. If we have one capacity and amount is set to 2, then we don't die, but if it's set to 1 we do?

File: Assets/Scripts/Boat/BoatCapacity.cs [hunk 2]
```diff
@@ -68,54 +126,71 @@ public int ResetCapacity()
...
-    public int IncreaseSouls(int loadToAdd)
+    public int IncreaseSouls(int loadToAdd=int.MaxValue)
     {
-        currentLoad += loadToAdd;
+        CurrentLoad += loadToAdd;
         return ReduceLoadToFitCapacity();
     }
 
+    //The set up event actions requires no return type and no parameters. This function just fulfills that requirement.
+    //TODO replace with Delegates might work?
```
Comment (TomasMorton) [inline Assets/Scripts/Boat/BoatCapacity.cs:162]: You could also do `GameStateManager.OnFerryingEnter += () => IncreaseSouls()`, but this is fine (it's clearer in its intent)

File: Assets/Scripts/Boat/BoatCapacity.cs [hunk 3]
```diff
@@ -34,19 +87,24 @@ void Start()
     /// <param name="amount">By how many units should the maximum capacity be increased.</param>
     public void IncreaseCapacity(int amount = 1)
```
Comment (baconeta) [inline Assets/Scripts/Boat/BoatCapacity.cs]: This will never get called without a parameter as `public void IncreaseCapacity()` will be called instead

**Review by baconeta — APPROVED** [review]
I made the appropriate bug fix changes and I now approve this message

**Pull request merged**